spec/ics-23-standard-accumulator/README.md

```diff
@@ -0,0 +1,228 @@
+
+## Definitions
+
+The *manager* of an accumulator is the actor with the ability and responsibility to add or remove items from the accumulator. Generally this will be the state machine of a blockchain.
+
+The *prover* is the actor responsible for generating accumulator proofs of inclusion or non-inclusion of particular elements. Generally this will be a relayer (see [ICS 18](../ics-18-relayer-algorithms)).
+
+The *verifier* is the actor who checks accumulator proofs in order to verify that the manager of the accumulator did or did not add a particular element. Generally this will be an IBC handler running on another chain.
+
+Accumulators are instantiated with a particular *element* type, which is assumed to be arbitrary serializable data.
+
+## Desired Properties
+
+This document only defines desired properties, not a concrete implementation — see "Properties" below.
+
+## Technical Specification
+
+### Datatypes
+
+An accumulator construction MUST specify the following datatypes, which are otherwise opaque (need not be introspected) but must be serializable:
+
+#### State
+
+An `AccumulatorState` is the full state of the accumulator, which will be stored by the manager.
+
+```golang
+type AccumulatorState struct
+```
+
+#### Root
+
+An `AccumulatorRoot` commits to a particular accumulator state and should be succinct.
+
+In certain accumulator constructions with succinct states, `AccumulatorState` and `AccumulatorRoot` may be the same type.
+
+```golang
+type AccumulatorRoot struct
+```
+
+#### Proof
+
+An `AccumulatorProof` demonstrates membership or non-membership for an element or set of elements, verifiable in conjunction with a known accumulator root. Proofs should be succinct.
+
+```golang
+type AccumulatorProof struct
+```
+
+### Required functions
+
+An accumulator construction MUST provide the following functions:
+
+#### Initialization
+
+The `generate` function initializes the state of the accumulator from an initial (possibly empty) set of elements.
+
+```coffeescript
+generate(Set<Element> initial) -> AccumulatorState
+```
+
+#### Root calculation
+
+The `calculateRoot` function calculates a succinct commitment to the accumulator state which can be used to verify proofs.
+
+```coffeescript
+calculateRoot(AccumulatorState state) -> AccumulatorRoot
+```
+
+#### Adding & removing elements
+
+The `add` function adds an element to an accumulator.
+
+```coffeescript
+add(AccumulatorState state, Element elem) -> AccumulatorState
+```
+
+The `remove` function removes an element from an accumulator.
+
+```coffeescript
+remove(AccumulatorState state, Element elem) -> AccumulatorState
+```
+
+#### Proof generation
+
+The `createMembershipWitness` function generates a proof that an element has been added to an accumulator.
```

Comment:
Why `createMembershipWitness` instead of `createMembershipProof`?

Reply:
`witness` is used in the literature but I guess we don't need to call it that

Reply:
If we use `proof` everywhere I think we should stick to it in the function's name

Reply:
Agreed, updated.